### PR TITLE
[Functionalization] detach_copy use set_ instead of _copy_from

### PR DESCRIPTION
**Pull request by alanwaketan**
Summary:
It looks like `_copy_from` somehow is not working correctly with PJRT on devices other than xla:0. Let's use set_ which also more loyally represents the semantics of detach.

Test Plan:
PJRT_DEVICE=TPU python test/pjrt/test_experimental_pjrt_tpu.py TestTpuCollectiveOps.test_broadcast_master_param_synchronized_parameters

### CODE REVIEW

**Comment by alanwaketan** [comment]
It looks like this change also happen to fix #4680.

**Comment by wonjoolee95** [comment]
> It looks like this change also happen to fix #4680. @wonjoolee95

Wow, really! That's amazing. Interesting.. wondering what was the original issue/error with the `PJRT` in the beginning?

**Review by wonjoolee95 — APPROVED** [review]
LGTM pending CI 🚀

**Comment by alanwaketan** [comment]
Thanks, Wonjoo.

File: test/spmd/test_xla_virtual_device.py [hunk 1]
```diff
@@ -48,7 +48,8 @@ def test_model_weight_metrics(self):
     model = nn.Linear(128, 64).to(xm.xla_device())
     xs.mark_sharding(model.weight, self._get_mesh((1, self.n_devices)),
                      partition_spec)
-    self.assertNotIn("VirtualDeviceUsage", met.counter_names())
+    self.assertIn("VirtualDeviceUsage", met.counter_names())
```
Comment (alanwaketan) [inline test/spmd/test_xla_virtual_device.py:51]: @steventk-g This change happens to restore the test results of virtual device back to pre-func period.

Comment (steventk-g) [reply test/spmd/test_xla_virtual_device.py:51]: Thanks for the heads up!

**Comment by steventk-g** [comment]
Added a TPU CI run to this branch to check the impact on SPMD

**Comment by alanwaketan** [comment]
> Added a TPU CI run to this branch to check the impact on SPMD

Interesting. May I ask how do you do that? Not seeing any changes in the PR or new labels?

**Comment by steventk-g** [comment]
> > Added a TPU CI run to this branch to check the impact on SPMD
> 
> Interesting. May I ask how do you do that? Not seeing any changes in the PR or new labels?

Go to the GCP project > Cloud Build Triggers, click "Run" on the trigger (all the way to the right in the row) and pass in the commit hash for HEAD of your branch.

File: test/spmd/test_xla_virtual_device.py (hunk 1 above)
Comment (yeounoh) [reply test/spmd/test_xla_virtual_device.py:51]: Thanks for the fix @alanwaketan

**Review by yeounoh — APPROVED** [review]
Thank you!

**Review by will-cromar — APPROVED** [review]
Thanks!
